src/mon/Monitor.cc

```diff
@@ -2271,6 +2271,120 @@ void Monitor::do_health_to_clog(bool force)
   health_status_cache.summary = summary;
 }
 
+namespace {
+  enum class scrubbed_or_deepscrubbed_t { SCRUBBED, DEEPSCRUBBED };
+
+  void print_unscrubbed_detailed(const std::pair<const pg_t,pg_stat_t> pg_entry,
+				 bool *section_opened,
+				 stringstream &ss,
+				 Formatter *f,
+				 scrubbed_or_deepscrubbed_t how_scrubbed) {
+
+    const auto& pg_stat(pg_entry.second);
+    if (!*section_opened) {
+      if (f != nullptr) {
+	f->open_array_section("unscrubbed_pgs");
+      }
+
+      *section_opened = true;
+    }
```

Comment:
I would put this in the caller loop, line 2358...  if ((not_scrubbed || not_deep_scrubbed) && !section_opened) { ... }